core/src/test/java/org/bitcoinj/crypto/HDPathTest.java

```diff
@@ -58,6 +58,22 @@ public void testExtendVarargs() {
         assertEquals("m/0H/1H/0H/1/0",  path5.toString());
     }
 
+    @Test
+    public void testParent() {
+        HDPath path1 = HDPath.parsePath("m/0H/1H");
+
+        assertEquals(HDPath.parsePath("m/0H"), path1.parent());
+
+        HDPath path2 = HDPath.parsePath("m/0H");
+
+        assertEquals(HDPath.parsePath(""), path2.parent());
+
+        HDPath path3 = HDPath.parsePath("");
+
+        assertEquals(HDPath.parsePath(""), path3.parent());
```

Comment:
We could also throw an exception in this case? What's the reason for making this edge case legal?

Reply:
I generally try to avoid exceptions and nulls, when possible. In this case, I think it make sense to return an empty list. In most use cases I expect the code will check before calling `.parent()`, but it is also possible to check for empty list after calling parent.

Reply:
If look at https://github.com/bitcoinj/bitcoinj/commit/15cccc34c820c41bc550b4343e76f2fff66805ec (lines 904-905 in the changed file) you'll see the check for `!path.isEmpty()` before calling `.parent()`.

Reply:
JavaDoc updated to explain this behavior more clearly.